go.mod

```diff
@@ -18,6 +18,8 @@ replace github.com/docker/docker => github.com/moby/moby v17.12.0-ce-rc1.0.20200
 // Client-go does not handle different versions of mergo due to some breaking changes - use the matching version
```

Comment:
Overall looks really good, I think this is close. Main concern is the cluster building. We are building the clusters like HTTPS/TLS clusters, which have minimal config. This leads to weird behavior like `curl https://foo --http2` negiotating `h2` ALPN, but then forwarding http/1.1 to the app which seems problematic

Reply:
Thinking `HttpProtocolOptions_AutoHttpConfig` could plausibly be used for this. Never used before in Istio though so may be suspicious.

I am ok merging in current state if we make it experimental behind an off-by-default flag (so we don't need to worry about breaking change when we fix it).

Reply:
hey @howardjohn , since we are getting too close to the release, would it be ok if we can create an issue for this and try to tackle it in future releases?
Based on your comment, I added the feature flag.
https://github.com/istio/istio/pull/36210/commits/54948eb40db0af6353dfdde4238486aee8cf24ab

Reply:
@howardjohn , I've added `hide-from-docs` for now